projects/ipdg_stokes/post_processing/solution_to_mesh_data_set.h

```diff
@@ -0,0 +1,48 @@
+#ifndef THESIS_POST_PROCESSING_SOLUTION_TO_MESH_DATA_SET_H
+#define THESIS_POST_PROCESSING_SOLUTION_TO_MESH_DATA_SET_H
+
+/**
+ * @file solution_to_mesh_data_set.h
+ * @brief Mappings from solution vectors of the LSE to mesh data sets
+ * representing physical properties
+ */
+
+#include <lf/assemble/dofhandler.h>
+#include <lf/mesh/mesh.h>
+#include <lf/mesh/utils/codim_mesh_data_set.h>
+
+#include <Eigen/Dense>
+
+namespace projects::ipdg_stokes {
+
+namespace post_processing {
+
+/**
+ * @brief Extract the basis function coefficients from the solution vector
+ * @param mesh A shared pointer to the corresponding mesh
+ * @param dofh The DOF handler for the used FE space
+ * @param solution The solution vector obtained from solving the PDE
+ * @returns A mesh data set mapping nodes of the mesh to the basis function
+ * coefficient associated with them
+ */
+lf::mesh::utils::CodimMeshDataSet<double> extractBasisFunctionCoefficients(
+    const std::shared_ptr<const lf::mesh::Mesh> &mesh,
+    const lf::assemble::DofHandler &dofh, const Eigen::VectorXd &solution);
+
+/**
+ * @brief Extract the flow velocity on cells from the solution vector
+ * @param mesh A shared pointer to the corresponding mesh
+ * @param dofh The DOF handler for the used FE space
+ * @param solution The solution vector obtained from solving the PDE
+ * @returns A mesh data set mapping cells of the mesh to the flow velocity on
+ * them
+ */
+lf::mesh::utils::CodimMeshDataSet<Eigen::Vector2d> extractVelocity(
+    const std::shared_ptr<const lf::mesh::Mesh> &mesh,
+    const lf::assemble::DofHandler &dofh, const Eigen::VectorXd &solution);
```

Comment:
Much nicer, especially when you're going to use first order basis functions, would be if you return a MeshFunction which can be evaluated on the cells of the mesh. Respectively you could write a new class that is a [MeshFunction](https://craffael.github.io/lehrfempp/group__mesh__function.html) and that represents the velocity anywhere on the mesh.